### PR TITLE
Implemented Hash_reputation and get_threat_notes actions

### PR DESCRIPTION
**Pull request by abhijeeta-metron**
Implemented following commands:
- Hash_reputation
- get_threat_notes
- add_threat_note

### CODE REVIEW

File: Apps/phsentinelone/sentinelone_connector.py [hunk 1]
```diff
@@ -817,6 +817,73 @@ def _handle_create_firewall_rule(self, param):
...
         return action_result.set_status(phantom.APP_SUCCESS, "Successfully created firewall rule")
 
+    def _handle_hash_reputation(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        hash = param.get('hash')
+        try:
+            site_ids = self._get_site_id(action_result)
+        except Exception:
+            return action_result.set_status(phantom.APP_ERROR, "Did not get proper response from the server")
```
Comment (devang-metron) [inline Apps/phsentinelone/sentinelone_connector.py]: site id param is not required, remove it

File: Apps/phsentinelone/sentinelone_connector.py [hunk 2]
```diff
@@ -817,6 +817,73 @@ def _handle_create_firewall_rule(self, param):
...
+        if phantom.is_fail(ret_val):
+            self.save_progress("Failed to get threat notes.  Error: {0}".format(action_result.get_message()))
+            return action_result.get_status()
+        action_result.add_data(response)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_add_note(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        s1_threat_ids = param['s1_threat_ids']
```
Comment (devang-metron) [inline Apps/phsentinelone/sentinelone_connector.py:855]: Isn't ids are in form of array?

File: Apps/phsentinelone/sentinelone_connector.py [hunk 3]
```diff
@@ -817,6 +817,73 @@ def _handle_create_firewall_rule(self, param):
...
+        summary['note'] = note
+        header = self.HEADER
+        header["Authorization"] = "APIToken %s" % self.token
+        try:
+            body = {
+                "data": {
+                            "text": note
+                        },
+                "filter": {
+                            "ids": s1_threat_ids,
```
Comment (devang-metron) [inline Apps/phsentinelone/sentinelone_connector.py:869]: You have to added multiple threats in form of array and not just string

**Review by devang-metron — CHANGES_REQUESTED** [review]

File: Apps/phsentinelone/sentinelone_connector.py (hunk 2 above)
Comment (abhijeeta-metron) [reply Apps/phsentinelone/sentinelone_connector.py:855]: yes

File: Apps/phsentinelone/sentinelone_connector.py (hunk 3 above)
Comment (abhijeeta-metron) [reply Apps/phsentinelone/sentinelone_connector.py:869]: Data type of ids is an array

**Review by devang-metron — APPROVED** [review]

File: Apps/phsentinelone/sentinelone_connector.py [hunk 4]
```diff
@@ -1051,6 +1111,12 @@ def handle_action(self, param):
             ret_val = self._handle_create_firewall_rule(param)
         elif action_id == 'hash_reputation':
             ret_val = self._handle_hash_reputation(param)
+        elif action_id == 'get_threat_notes':
+            ret_val = self._handle_get_threat_notes(param)
+        elif action_id == 'add_note':
```
Comment (devang-metron) [inline Apps/phsentinelone/sentinelone_connector.py]: add_note doesn't depict to which thing to add note to.

Comment (devang-metron) [reply Apps/phsentinelone/sentinelone_connector.py]: also change function name _handle_add_note for the same.

**Review by devang-metron — CHANGES_REQUESTED** [review]

**Review by devang-metron — APPROVED** [review]

**Pull request merged**